modules/lang-expression/src/main/java/org/elasticsearch/script/expression/ReplaceableConstDoubleValueSource.java

```diff
@@ -38,8 +39,12 @@ public boolean needsScores() {
 
     @Override
     public Explanation explain(LeafReaderContext ctx, int docId, Explanation scoreExplanation) throws IOException {
-        if (fv.advanceExact(docId)) return Explanation.match((float) fv.doubleValue(), "ReplaceableConstDoubleValues");
-        else return Explanation.noMatch("ReplaceableConstDoubleValues");
+        // TODO where is this explain called? I bet it's never tested, and probably never called.
+        ReplaceableConstDoubleValues fv = specialValues.get(ctx);
+        if (fv.advanceExact(docId)) {
+            return Explanation.match((float) fv.doubleValue(), "ReplaceableConstDoubleValues");
+        }
+        return Explanation.noMatch("ReplaceableConstDoubleValues");
```

Comment:
How do aggregation scripts get explained? I am thinking this may never get called. If anyone knows how this gets called, that would help me figure out how to test this, otherwise I am thinking of throwing UnsupportedOperationException in this method, just to surface that it is never called.

Reply:
The class is also used for score scripts, so it does need to implement `explain` I think.

Reply:
can you show me where? I can't find that usage.

Reply:
https://github.com/elastic/elasticsearch/blob/2998547c1d846ef99e1b7b02854cd42b34220071/modules/lang-expression/src/main/java/org/elasticsearch/script/expression/ExpressionScriptEngine.java#L374C15-L374C15

Reply:
Thanks. I am perplexed, because the score usage does not pass the replaceable around, which makes it not replaceable  as setValue will never be called. Effectively, will the value not always be `0`? Maybe @jdconrad can help us here :)

Reply:
I tried replacing the replaceable usage in scoring with a constant and I got no failures: does it mean that it's not needed or that we have no test coverage for it? Also, if it is a constant (0) do we even need to add it to the bindings. All in all, I am thinking that we could clean things up and get rid of the explain here, as I don't find a way to test my changes to it.

Reply:
I believe it should never be called for score scripts, since it is specifically for _value, which is not a thing there. However, I _vaguely_ recall some issue when migrating to score script context, which caused me to add _value to score scripts.

Could you try throwing UnsupportedOperationException and see if anything fails?

Reply:
It looks like this will never be called by anything. From what I can tell explain will only work on a script score query using Painless.

Reply:
thanks a lot for looking into this @jdconrad & @rjernst . I will try to clean this up further as a follow-up, possibly removing the explain impl if never called.